src/test/java/com/google/sps/SentimentServletTest.java

```diff
@@ -23,48 +28,71 @@
   @Mock
   HttpServletRequest requestMock;
 
+  @Spy
+  HttpServletResponse responseMock;
+
   /**
-   * Takes a Json String of an array and returns the elements of the array as a String, separated
-   * by periods.
-   * Ex. "["foo","bar"]" -> "foo. bar"
+   * A Json String in the form of an array should be converted to a String of period separated
+   * comments. A sentiment score should be calculated and the score should be printed in a Json
+   * object. 
+   * Ex. "foo" -> "{"commentScore":X.XX}" where X.XX is a float, for any # of decimal places. 
    */
   @Test
-  public void jsonArrayConvertsToPeriodSeparatedString() throws IOException {
+  public void jsonArrayPrintsVideoAnalysisJsonObject() throws IOException {
     String json = "[\"a\",\"b\",\"c\"]"; // "["a","b","c"]"
     when(requestMock.getReader()).thenReturn(new BufferedReader(new StringReader(json)));
     
-    String actual = sentimentServlet.requestToString(requestMock);
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(responseMock.getWriter()).thenReturn(writer);
+
+    new SentimentServlet().doGet(requestMock, responseMock);
 
-    Assert.assertEquals("a. b. c", actual);
+    String pattern = "\\{\"commentScore\":[-+]?[0-9]*\\.?[0-9]+\\}\n";
```

Comment:
Is this unit test actually getting a response from the sentiment API which is why you are using regex to catch any score that it returns?

Unit tests should only test the direct file they are testing. You are technically doing this by doing the regex and not caring about the actual score but if its actually making a service call then your unit test might fail if the service goes down. Instead, you should mock with a hardcoded value what the sentiment API returns and in your unit test you can verify that the value the API returned was correctly translated into your servlet response.